### ISSUE TITLE
Integer display width breaks diff on MySQL 8->8.0.19+ upgrade

### ISSUE THREAD

**Issue by trevorr** [skeema/skeema#158]
**Describe the bug**
After upgrading from MySQL 8.0.15 to 8.0.22, Skeema skips all my tables (which all have integer columns): ``Skipping table `xxx`: unable to generate DDL due to use of unsupported features.``

**Environment**
- Skeema version: 1.4.7, commit df922ef8cbc92f7a3c11a24790aa503970b7a237, released 2021-01-05T23:36:31Z
- Database vendor, version, OS/platform: mysql:8.0.22 Docker Hub image

**To Reproduce**
```
$ skeema diff localhost --debug
2021-02-06 14:01:22 [INFO]  Generating diff of 127.0.0.1:3306 xxx vs xxx/*.sql
2021-02-06 14:01:22 [WARN]  Skipping table `xxx`: unable to generate DDL due to use of unsupported features. Use --debug for more information.
2021-02-06 14:01:22 [DEBUG] --- Expected CREATE
2021-02-06 14:01:22 [DEBUG] +++ MySQL-actual SHOW CREATE
2021-02-06 14:01:22 [DEBUG] @@ -2,2 +2,2 @@
2021-02-06 14:01:22 [DEBUG] -  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
2021-02-06 14:01:22 [DEBUG] -  `source_id` int(10) unsigned NOT NULL,
2021-02-06 14:01:22 [DEBUG] +  `id` int unsigned NOT NULL AUTO_INCREMENT,
2021-02-06 14:01:22 [DEBUG] +  `source_id` int unsigned NOT NULL,
```

**Expected behavior**
I would expect the diff to succeed and show no changes. It does so when running against 8.0.15 with the same schema.

**Additional context**
Skeema diffs have been generating this warning in MySQL 8: `Warning: #1681 Integer display width is deprecated and will be removed in a future release.` Integer display width has been [deprecated](https://dev.mysql.com/worklog/task/?id=13127) for awhile, but it seems it was only removed from `SHOW CREATE` in a recent version.

**Comment by evanelias** [comment]
Thank you for the issue report! I believe this only occurs if you specifically upgrade from a pre-8.0.19 version of MySQL 8, to 8.0.19+. It doesn't occur if you start with 8.0.19+, which is why Skeema's integration tests don't catch it -- the tests always run against the latest 8.0, but as a fresh install rather than a upgrade. It also won't happen if you upgrade from pre-8 (e.g. 5.6 or 5.7) directly to 8.0.19+ without ever using a pre-8.0.19 version of MySQL 8.

The root cause is that in this specific upgrade scenario, `information_schema` retains the display widths but `SHOW CREATE TABLE` omits them, which confuses one of Skeema's safety checks. On closer inspection the situation is actually described in MySQL 8.0.19's [release notes](https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-19.html#mysqld-8-0-19-feature) - see last paragraph of third bullet.

I hope to commit a Skeema fix within a few days. The fix itself should be straightforward, although the test coverage might be a bit trickier.

> Skeema diffs have been generating this warning in MySQL 8

That's related but I believe it's a separate root cause. If you remove the display widths from your CREATE statements in your *.sql files, do the warnings stop?

**Comment by evanelias** [comment]
A fix for this bug has been committed to the main branch. It will be included in the next Skeema patch release, likely either later this week or early next week. Thanks again for the issue report!

btw if you need an urgent work-around: if you manually ALTER a table once outside of Skeema, it appears that MySQL will clean up its internal data dictionary for that table and start omitting display widths from `information_schema` after that one ALTER. This will allow Skeema v1.4.7 to handle subsequent diffs and alterations for the table. However, you would need to repeat this process for every affected table.

**Comment by evanelias** [comment]
This fix has been included in today's release of [Skeema v1.5.0](https://github.com/skeema/skeema/releases/tag/v1.5.0). Thanks again for the bug report!
